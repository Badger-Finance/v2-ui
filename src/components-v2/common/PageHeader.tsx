import React from 'react';
import { Typography, makeStyles } from '@material-ui/core';
import WalletWidget from './WalletWidget';
import GasWidget from './GasWidget';

import { getNetwork } from '../../mobx/utils/helpers';

export interface PageHeaderProps {
	title: string;
	subtitle?: string;
}

const useStyles = makeStyles((theme) => ({
	headerWidgets: {
		display: 'flex',
		alignItems: 'center',
		[theme.breakpoints.down('sm')]: {
			justifyContent: 'center',
		},
	},
	headerContainer: {
		display: 'flex',
		marginTop: theme.spacing(3),
		[theme.breakpoints.down('sm')]: {
			marginTop: theme.spacing(8),
		},
		[theme.breakpoints.down('xs')]: {
			flexDirection: 'column',
		},
		alignItems: 'center',
		justifyContent: 'space-between',
	},
	headerContent: {
		[theme.breakpoints.down('xs')]: {
			marginBottom: theme.spacing(1),
		},
	},
}));

const PageHeader: React.FC<PageHeaderProps> = (props: PageHeaderProps) => {
	const classes = useStyles();
	const { title, subtitle } = props;
	return (
		<div className={classes.headerContainer}>
			<div className={classes.headerContent}>
				<Typography variant="h2" color="textPrimary">
					{title}
				</Typography>
				{subtitle && (
					<Typography variant="subtitle1" color="textSecondary">
						{subtitle}
					</Typography>
				)}
			</div>
			<div className={classes.headerWidgets}>
				<GasWidget />
				<WalletWidget />
			</div>
		</div>
	);
};

export default PageHeader;
