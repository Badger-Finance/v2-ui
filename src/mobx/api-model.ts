/**
 * All API Model Objects should be directly in line with API interface model
 * objects to ensure data compatiblity. Interfaces located at link below:
 * https://github.com/axejintao/badger-api/tree/staging/src/interface
 */
import BigNumber from "bignumber.js";

export interface Geyser {
  emissions: Emission[],
}

export interface Token {
  address: string;
  name: string;
  symbol: string;
  decimals: number;
}

export interface Emission {
  token: Token,
  unlockSchedule: UnlockSchedule,
}

export interface UnlockSchedule {
	initialLocked: BigNumber,
	endAtSec: BigNumber,
	durationSec: BigNumber,
	startTime: BigNumber,
}

export interface TokenBalance {
  address: string;
  name: string;
  symbol: string;
  decimals: number;
  balance: number,
  value: number,
}

export interface ValueSource {
  name: string,
  apy: number,
  performance: Performance,
}

export interface Sett {
  name: string,
  asset: string,
  ppfs: number,
  value: number,
  apy: number,
  tokens: TokenBalance[],
  sources: ValueSource[],
  geyser?: Geyser,
}

export interface SettSummary {
  name: string,
  asset: string,
  value: number,
  tokens: TokenBalance[],
}
