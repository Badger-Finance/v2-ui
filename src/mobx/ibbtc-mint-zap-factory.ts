import { IbBTCMintZap } from './model/vaults/ibbtc-mint-zap';
import { RootStore } from './RootStore';
import addresses from 'config/ibBTC/addresses.json';
import { TokenZap } from './model/vaults/token-zap';
import { GeneralVaultZap } from './model/vaults/general-vault-zap';
import { RenVaultZap } from './model/vaults/ren-vault-zap';
import { Token } from '@badger-dao/sdk';

export class IbBTCMintZapFactory {
  private static tokenZaps: Record<string, TokenZap> = {};
  private static generalVaultZaps: Record<string, GeneralVaultZap> = {};
  private static renVaultZaps: Record<string, RenVaultZap> = {};

  static getIbBTCZap(store: RootStore, token: Token): IbBTCMintZap {
    if (addresses.mainnet.contracts.TokenZap.supportedTokens.includes(token.address)) {
      if (!this.tokenZaps[token.address]) {
        this.tokenZaps[token.address] = new TokenZap(store, token);
      }
      return this.tokenZaps[token.address];
    }

    if (addresses.mainnet.contracts.GeneralVaultZap.supportedTokens.includes(token.address)) {
      if (!this.generalVaultZaps[token.address]) {
        this.generalVaultZaps[token.address] = new GeneralVaultZap(store, token);
      }
      return this.generalVaultZaps[token.address];
    }

    if (addresses.mainnet.contracts.RenVaultZap.supportedTokens.includes(token.address)) {
      if (!this.renVaultZaps[token.address]) {
        this.renVaultZaps[token.address] = new RenVaultZap(store, token);
      }
      return this.renVaultZaps[token.address];
    }

    throw new Error(`Unrecognized peak type for token => ${token.symbol}`);
  }
}
