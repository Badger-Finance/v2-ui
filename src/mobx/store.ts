import { RouterStore } from 'mobx-router';
import UiState from './reducers';
import WalletStore from './stores/walletStore';
import ContractsStore from './stores/contractsStore';
import AirdropStore from './stores/airdropStore';
import RebaseStore from './stores/rebaseStore';
import RewardsStore from './stores/rewardsStore';
import IbBTCStore from './stores/ibBTCStore';
import TransactionsStore from './stores/transactionsStore';
import SettStoreV2 from './stores/settStoreV2';
import { NETWORK_LIST } from '../config/constants';
import HoneyPotStore from './stores/honeyPotStore';

export class RootStore {
	public router: RouterStore<RootStore>;
	public wallet: WalletStore;
	public uiState: UiState;
	public contracts: ContractsStore;
	public airdrops: AirdropStore;
	public rebase: RebaseStore;
	public rewards: RewardsStore;
	public ibBTCStore: IbBTCStore;
	public setts: SettStoreV2;
	public transactions: TransactionsStore;
	public honeyPot: HoneyPotStore;

	constructor() {
		this.router = new RouterStore<RootStore>(this);
		this.wallet = new WalletStore(this);
		this.contracts = new ContractsStore(this);
		this.airdrops = new AirdropStore(this);
		this.rebase = new RebaseStore(this);
		this.rewards = new RewardsStore(this);
		this.uiState = new UiState(this);
		this.ibBTCStore = new IbBTCStore(this);
		this.honeyPot = new HoneyPotStore(this);
		// RenVM transactions store.
		this.transactions = new TransactionsStore(this);
		this.setts = new SettStoreV2(this);

		this.walletRefresh();
	}

	async walletRefresh(): Promise<void> {
		const chain = this.wallet.network.name;
		const refreshData = [
			this.setts.loadAssets(chain),
			this.setts.loadPrices(chain),
			this.wallet.getGasPrice(),
			this.contracts.updateProvider(),
		];
		if (chain === NETWORK_LIST.ETH) {
			refreshData.push(this.setts.loadGeysers(chain));
			refreshData.push(this.rebase.fetchRebaseStats());
		} else {
			refreshData.push(this.setts.loadSetts(chain));
		}
		await Promise.all(refreshData);

		if (this.wallet.connectedAddress) {
			this.contracts.updateProvider();
			await this.wallet.getGasPrice();
			await this.contracts.fetchContracts();
			if (chain === NETWORK_LIST.ETH) {
				this.uiState.reduceRebase();
				this.ibBTCStore.init();
				this.rewards.fetchSettRewards();
				this.uiState.reduceTreeRewards();
				this.airdrops.fetchAirdrops();
				this.uiState.reduceAirdrops();
			}
			this.uiState.reduceStats();
		}
	}
}

const store = new RootStore();

export default store;
