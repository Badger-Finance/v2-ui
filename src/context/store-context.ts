import { createContext } from "react";
import { RootStore } from "../mobx/store";

// Use React context to make your store available in your application
export const StoreContext = createContext({} as RootStore);
export const StoreProvider = StoreContext.Provider;