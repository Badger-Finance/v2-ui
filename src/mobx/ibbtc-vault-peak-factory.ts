import { IbbtcVaultPeak } from './model/vaults/ibbtc-vault-peak';
import { RootStore } from './store';
import addresses from 'config/ibBTC/addresses.json';
import { ZapPeak } from './model/vaults/zap-peak';
import { YearnPeak } from './model/vaults/yearn-peak';
import { BadgerPeak } from './model/vaults/badger-peak';
import { IbbtcOptionToken } from './model/tokens/ibbtc-option-token';

export class IbbtcVaultPeakFactory {
	static createIbbtcVaultPeakForToken(store: RootStore, token: IbbtcOptionToken): IbbtcVaultPeak {
		if (addresses.mainnet.contracts.ZapPeak.supportedTokens.includes(token.symbol)) {
			return new ZapPeak(store, token);
		}

		if (addresses.mainnet.contracts.yearnWBTCPeak.supportedTokens.includes(token.symbol)) {
			return new YearnPeak(store, token);
		}

		if (addresses.mainnet.contracts.BadgerSettPeak.supportedTokens.includes(token.symbol)) {
			return new BadgerPeak(store, token);
		}

		throw new Error(`Unrecognized peak type for token => ${token.symbol}`);
	}
}
