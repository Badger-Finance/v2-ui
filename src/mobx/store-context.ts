import { RootStore } from './store';
import { createContext } from 'react';

// Use React context to make your store available in your application
export const StoreContext = createContext({} as RootStore);
export const StoreProvider = StoreContext.Provider;
