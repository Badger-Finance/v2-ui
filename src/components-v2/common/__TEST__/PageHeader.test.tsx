import React from 'react';
import PageHeader from '../PageHeader';
import store from '../../../mobx/store';
import '@testing-library/jest-dom';
import { StoreProvider } from 'mobx/store-context';
import { render } from '@testing-library/react';

test('Renders correctly', () => {
	const title = 'Test title';
	const subtitle = 'Test subtitle';
	const { container } = render(
		<StoreProvider value={store}>
			<PageHeader title={title} subtitle={subtitle} />
		</StoreProvider>,
	);
	expect(container).toMatchSnapshot();
});
