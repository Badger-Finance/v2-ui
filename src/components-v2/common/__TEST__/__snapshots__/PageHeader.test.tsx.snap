// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly 1`] = `
<div>
  <div
    class="makeStyles-headerContainer-2"
  >
    <div
      class="makeStyles-headerContent-3"
    >
      <h2
        class="MuiTypography-root MuiTypography-h2 MuiTypography-colorTextPrimary"
      >
        Test title
      </h2>
      <h6
        class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-colorTextSecondary"
      >
        Test subtitle
      </h6>
    </div>
    <div
      class="makeStyles-headerWidgets-1"
    >
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorSecondary MuiOutlinedInput-colorSecondary makeStyles-gasSelector-4 MuiInputBase-adornedStart MuiOutlinedInput-adornedStart"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          style="cursor: pointer; font-size: 1.2rem; margin-right: .8rem;"
          viewBox="0 0 24 24"
        >
          <path
            d="M19.77 7.23l.01-.01-3.72-3.72L15 4.56l2.11 2.11c-.94.36-1.61 1.26-1.61 2.33 0 1.38 1.12 2.5 2.5 2.5.36 0 .69-.08 1-.21v7.21c0 .55-.45 1-1 1s-1-.45-1-1V14c0-1.1-.9-2-2-2h-1V5c0-1.1-.9-2-2-2H6c-1.1 0-2 .9-2 2v16h10v-7.5h1.5v5c0 1.38 1.12 2.5 2.5 2.5s2.5-1.12 2.5-2.5V9c0-.69-.28-1.32-.73-1.77zM12 10H6V5h6v5zm6 0c-.55 0-1-.45-1-1s.45-1 1-1 1 .45 1 1-.45 1-1 1z"
          />
        </svg>
        <div
          aria-haspopup="listbox"
          class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedStart MuiOutlinedInput-inputAdornedStart"
          role="button"
          tabindex="0"
        >
          75
        </div>
        <input
          aria-hidden="true"
          class="MuiSelect-nativeInput"
          tabindex="-1"
          value="standard"
        />
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 10l5 5 5-5z"
          />
        </svg>
        <fieldset
          aria-hidden="true"
          class="PrivateNotchedOutline-root-5 MuiOutlinedInput-notchedOutline"
          style="padding-left: 8px;"
        >
          <legend
            class="PrivateNotchedOutline-legend-6"
            style="width: 0.01px;"
          >
            <span>
              â€‹
            </span>
          </legend>
        </fieldset>
      </div>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-walletButton-12 MuiButton-containedSecondary MuiButton-disableElevation"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Click to connect
          <span
            class="MuiButton-endIcon MuiButton-iconSizeMedium"
          >
            <div
              class="makeStyles-walletDot-9 makeStyles-redDot-10"
            />
          </span>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;
